src/App.tsx

```diff
@@ -1,50 +1,107 @@
-import React from 'react';
+import React, { useState } from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
+import cn from 'classnames';
+import { GoodsList } from './components/GoodList/GoodList';
+import { goodsFromServer } from './api/data';
 
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+interface SortTypes {
+  ALPHA: string;
+  LENGTH: string;
+  NONE: string;
+}
+
+const sortFields: SortTypes = {
+  ALPHA: 'alphabet',
+  LENGTH: 'length',
+  NONE: 'string',
+};
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  isReversedField: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case sortFields.ALPHA:
+          return a.localeCompare(b);
+
+        case sortFields.LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversedField) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversedField, setIsReversedField] = useState(false);
+  const list = getPreparedGoods(goodsFromServer, sortField, isReversedField);
```

Comment:
```suggestion
  const preparedGods = getPreparedGoods(goodsFromServer, sortField, isReversedField);
```
according to function name